shipment_advice_planner_toursolver/models/toursolver_delivery_window.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import math
+
+from psycopg2.extensions import AsIs
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ToursolverDeliveryWindow(models.Model):
```

Comment:
to be removed with the use of `base_time_window` (moreover the `preference` field is never used)